graylog2-server/src/main/java/org/graylog2/streams/filters/StreamDestinationFilterService.java

```diff
@@ -66,13 +67,17 @@ public class StreamDestinationFilterService {
     private final MongoPaginationHelper<StreamDestinationFilterRuleDTO> paginationHelper;
     private final MongoUtils<StreamDestinationFilterRuleDTO> utils;
     private final ClusterEventBus clusterEventBus;
+    private final Map<String, DestinationFilterActionGuard> destinationFilterLicenseCheckMap;
```

Comment:
Sorry if I am asking dumb question, but is the whole complexity with multi-binders and map injection really needed, if this seems to be the only place that need this check, that uses `DestinationFilterActionGuard`?

Isn't it that in fact in lines 135-143 you just want to check if a user has a enterprise license?

Reply:
@luk-kaminski  afaik we don't do license checks in graylog open, and this should only concern data warehouse filters. I guess the plan is to add more target types in the future

Reply:
@luk-kaminski According to that reasoning, everything we plugged in is executed behind a complex chain of invocations in server.
It's of no concern to the user in open that in enterprise we are performing a license check for this type of output (which is also only available in enterprise) and the specific action. We could have other conditions as well guarding the execution of actions. Besides that, I don't see how we ever brought in enterprise functionality without interfacing it. 
Pulling license check into graylog open doesn't seem to be the way to go, having a single class in open that is aware of an enterprise implementation also doesn'T seem to be the way to go. 
So what would you suggest instead?

Reply:
Without digging too much into details, if I were to try to imagine how to do it, I'd probably consider something like this:
I'd use OptionalBinder for DataWarehouseLicenseCheck, with default implementation in gl-server returning always `false`. 
I'd add proper binding in enterprise to implementation that checks license. 
I'd throw away the whole DestinationFilterActionGuard concept as long as we have only one action, as a unnecessary step/layer in the invocation chain. 
I'd throw away multibinding. 

As you see, it is not about putting the license check in the open code.

I am also not aware of the context of this PR. 
If it is badly needed or you are in a hurry, we could consider merging it as it is an leave possible simplification for cooldown (or ignoring simplification if further development needs that action-based layer of abstraction).

Reply:
Removed the enum and introduced a validator for only that single action. I am still throwing an exception so that I can return a bit more information back to the user